test/upgrade/2.0.0-2.1.0.js

```diff
@@ -0,0 +1,1108 @@
+      authTokenContractENS: await configHandler.connect(rando).getAuthTokenContract(AuthTokenType.ENS),
+      maxExchangesPerBatch: await configHandler.connect(rando).getMaxExchangesPerBatch(),
+      maxRoyaltyPecentage: await configHandler.connect(rando).getMaxRoyaltyPecentage(),
+      maxResolutionPeriod: await configHandler.connect(rando).getMaxResolutionPeriod(),
+      minDisputePeriod: await configHandler.connect(rando).getMinDisputePeriod(),
+      accessControllerAddress: await configHandler.connect(rando).getAccessControllerAddress(),
+    };
+  }
+
+  async function getDisputeContractState() {
+    let disputesState = [];
+    let disputesStatesState = [];
+    let disputeTimeoutState = [];
+    let isDisputeFinalizedState = [];
+    for (let id = 1; id <= exchanges.length; id++) {
+      disputesState.push(await disputeHandler.connect(rando).getDispute(id));
+      disputesStatesState.push(await disputeHandler.connect(rando).getDisputeState(id));
+      disputeTimeoutState.push(await disputeHandler.connect(rando).getDisputeTimeout(id));
+      isDisputeFinalizedState.push(await disputeHandler.connect(rando).isDisputeFinalized(id));
+    }
+
+    return { disputesState, disputesStatesState, disputeTimeoutState, isDisputeFinalizedState };
+  }
+
+  async function getFundsContractState() {
+    // all id count
+    const totalCount = DRs.length + sellers.length + buyers.length + agents.length;
+    let groupsState = [];
+
+    // Query even the ids where it's not expected to get the entity
+    for (let id = 1; id <= totalCount; id++) {
+      groupsState.push(await fundsHandler.connect(rando).getAvailableFunds(id));
+    }
+
+    return { groupsState };
+  }
+
+  async function getGroupContractState() {
+    let groupsState = [];
+    for (let id = 1; id <= groups.length; id++) {
+      groupsState.push(await groupHandler.connect(rando).getGroup(id));
+    }
+
+    let nextGroupId = await groupHandler.connect(rando).getNextGroupId();
+    return { groupsState, nextGroupId };
+  }
+
+  async function getTwinContractState() {
+    let twinsState = [];
+    for (let id = 1; id < twins.length; id++) {
+      twinsState.push(await twinHandler.connect(rando).getTwin(id));
+    }
+
+    let nextTwinId = await twinHandler.connect(rando).getNextTwinId();
+    return { twinsState, nextTwinId };
+  }
+
+  async function getMetaTxContractState() {
+    return {};
+  }
+
+  const paddingType = {
+    NONE: 0,
+    START: 1,
+    END: 2,
+  };
+
+  function getMappinStoragePosition(slot, key, padding = paddingType.NONE) {
+    let keyBuffer;
+    switch (padding) {
+      case paddingType.NONE:
+        keyBuffer = ethers.utils.toUtf8Bytes(key);
+        break;
+      case paddingType.START:
+        keyBuffer = Buffer.from(ethers.utils.hexZeroPad(key, 32).toString().slice(2), "hex");
+        break;
+      case paddingType.END:
+        keyBuffer = Buffer.from(key.slice(2).padEnd(64, "0"), "hex"); // assume key is prefixed with 0x
```

Comment:
Why is this padding to 64?

Reply:
I'm padiding string directly, not bytes. If I first converted it to bytes and pad then, then it'd be 32.
I'm working with some assumptions here (like "key" is hex string, prefixed with "0x). For current use it's seems ok and I also don't expect problems in the future (the only types that are end-paded are bytesXX types with XX<32).